File: api/src/main/java/com/api/app/entity/PayBase.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("PayBase")
@Getter
@Setter
public class PayBase implements Serializable {
    private static final long serialVersionUID = 5678901234567890123L;

    @Schema(description = "결제번호")
    private String payNo;

    @Schema(description = "결제유형코드")
    private String payTypeCode;

    @Schema(description = "결제방식코드")
    private String payWayCode;

    @Schema(description = "결제상태코드")
    private String payStatusCode;

    @Schema(description = "승인번호")
    private String approveNo;

    @Schema(description = "주문번호")
    private String orderNo;

    @Schema(description = "클레임번호")
    private String claimNo;

    @Schema(description = "상위결제번호")
    private String upperPayNo;

    @Schema(description = "거래번호")
    private String trdNo;

    @Schema(description = "결제완료일시")
    private LocalDateTime payFinishDateTime;

    @Schema(description = "회원번호")
    private String memberNo;

    @Schema(description = "결제금액")
    private Long amount;

    @Schema(description = "취소가능금액")
    private Long cancelableAmount;
}

File: api/src/main/kotlin/com/api/app/entity/PayBase.kt
package com.api.app.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "PAY_BASE")
class PayBase : SystemEntity() {

    @Id
    @Column(name = "PAY_NO", length = 15, nullable = false)
    var payNo: String = ""

    @Column(name = "PAY_TYPE_CODE", nullable = false, length = 3)
    var payTypeCode: String = ""

    @Column(name = "PAY_WAY_CODE", nullable = false, length = 3)
    var payWayCode: String = ""

    @Column(name = "PAY_STATUS_CODE", nullable = false, length = 3)
    var payStatusCode: String = ""

    @Column(name = "APPROVE_NO", length = 50)
    var approveNo: String? = null

    @Column(name = "ORDER_NO", length = 15)
    var orderNo: String? = null

    @Column(name = "CLAIM_NO", length = 15)
    var claimNo: String? = null

    @Column(name = "UPPER_PAY_NO", length = 15)
    var upperPayNo: String? = null

    @Column(name = "TRD_NO", length = 100)
    var trdNo: String? = null

    @Column(name = "PAY_FINISH_DATE_TIME")
    var payFinishDateTime: LocalDateTime? = null

    @Column(name = "MEMBER_NO", nullable = false, length = 15)
    var memberNo: String = ""

    @Column(name = "AMOUNT", nullable = false, columnDefinition = "numeric")
    var amount: Long = 0

    @Column(name = "CANCELABLE_AMOUNT", columnDefinition = "numeric")
    var cancelableAmount: Long? = null

    @Column(name = "PG_TYPE_CODE", length = 3)
    var pgTypeCode: String? = null
}
